### PR TITLE
search: support global search for symbols

### PR DESCRIPTION
**Pull request by stefanhengl**
We have neglected symbol search in our optimizations for global search
so far. With this change we skip repo resolution for symbol search just like
we do for path and content queries.

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]
LGTM

File: internal/search/types.go [hunk 1]
```diff
@@ -110,16 +110,16 @@ const (
...
-	// (2) when a query does not specify any patterns for a content or file path search.
-	SkipContentAndPathSearch
+	// (2) when a query does not specify any patterns, include patterns, or exclude pattern.
+	SkipUnindexed
 )
 
 var globalSearchModeStrings = map[GlobalSearchMode]string{
-	ZoektGlobalSearch:        "ZoektGlobalSearch",
-	SearcherOnly:             "SearcherOnly",
-	SkipContentAndPathSearch: "SkipContentAndPathSearch",
```
Comment (rvantonder) [inline internal/search/types.go:122]: This change and related ones looks like a pure rename. It would make review life wonderful 😄  if there are separate commits (doesn't have to be separate PRs) for pure rename changes like these. Then it's easy to focus on the meat of the changes.

File: internal/search/symbol/symbol.go [hunk 2]
```diff
@@ -57,29 +66,30 @@ func Search(ctx context.Context, args *search.TextParameters, limit int, stream
...
+					cancel()
+					run.Error(err)
+				}
 			}
-		}
-	})
+		})
+	}
 
-	for _, repoRevs := range indexed.Unindexed {
```
Comment (rvantonder) [inline internal/search/symbol/symbol.go:82]: Missed this one! I guess we can unexport `Unindexed` to discover if there are any others.

**Review by rvantonder — APPROVED** [review]
Gonna be cool to see that search_blitz graph drop :-)

**Comment by keegancsmith** [comment]
> Gonna be cool to see that search_blitz graph drop :-)

will have to wait patiently for summer so our code can unfreeze.

**Comment by rvantonder** [comment]
@keegancsmith you're a poet and didn't know it

(or maybe you did :))

**Comment by keegancsmith** [comment]
Shall I compare thee to a summer's day? Thou art more lovely and more temperate. Rough winds do shake the darling buds of our next release.

**Comment by rvantonder** [comment]
LOL. OK let me try, umm...

discount-tent
